presto-main/src/main/java/com/facebook/presto/sql/planner/Partitioning.java

```diff
@@ -162,11 +170,16 @@ public boolean isPartitionedOn(Collection<Symbol> columns, Set<Symbol> knownCons
     {
         // partitioned on (k_1, k_2, ..., k_n) => partitioned on (k_1, k_2, ..., k_n, k_n+1, ...)
         // can safely ignore all constant columns when comparing partition properties
-        return arguments.stream()
+        // If the partitioning includes expression, it's not partitioned on any set of columns / symbols.
+        boolean partitioningIncludesExpression = arguments.stream()
```

Comment:
Since we don't have `SymbolReference` as expression for now, currently `ArgumentBinding.isColumnReference()` would mean the same as `ArgumentBinding.isVariable()` right? So do we need to add an `isColumnReference()` to `ArgumentBinding`?